src/report/annotation.go

```diff
@@ -115,15 +107,15 @@ func sortSeverities(severityCounts map[string]int32) []string {
 }
 
 // sort severity strings by rank, then alphabetically
-func compareSeverities(a, b string) int {
-	rank := rankSeverity(a) - rankSeverity(b)
+func compareSeverities(a, b types.FindingSeverity) int {
+	rank := rankSeverity(string(a)) - rankSeverity(string(b))
 
 	if rank != 0 {
 		return rank
 	}
 
 	// for unknown severities, sort alphabetically
-	return strings.Compare(a, b)
+	return strings.Compare(string(a), string(b))
 }
 
 func rankSeverity(s string) int {
```

Comment:
nit: this is used to rank severities, not strings, so why not use the `types.FindingSeverity`?

```suggestion
func rankSeverity(s types.FindingSeverity) int {
```

With this, the change to the function call can be reverted:

```diff
 func compareSeverities(a, b types.FindingSeverity) int {
-  rank := rankSeverity(string(a)) - rankSeverity(string(b))
+  rank := rankSeverity(a) - rankSeverity(b)
```

Reply:
applied